client/ConnectionThread.py

```diff
@@ -0,0 +1,109 @@
+            except socket.error:
+                if i == 5:
+                    print("Couldn't send message. Receiver must've closed connection!")
+                    self.handler_status.is_connected = -1
+        self.lock.release()
+
+    def get_counter(self):
+        self.lock.acquire()
+        for i in range(5):
+            try:
+                msg = self.conn.recv(2048)
+                msg = msg.decode()
+                if msg == '' or msg == ' ':
+                    print("Socket connection shut down")
+                    self.handler_status.is_connected = -1
+                print(msg)
+                break
+            except socket.error:
+                if i == 5:
+                    print("Couldn't receive message. Receiver must've closed connection!")
+                    self.handler_status.is_connected = -1
+        self.lock.release()
+
+    def connect(self):
+        tries = 0
+        self.socket.settimeout(10.0)
+        print(f"Trying to connect with: {self.handler_status.player_ip}:{self.handler_status.player_port}")
+        while True:
+            try:
+                self.socket.connect((self.handler_status.player_ip, self.handler_status.player_port))
+                self.socket.settimeout(None)
+                self.conn = self.socket
+                print(f"Connected with: {self.handler_status.player_ip}:{self.handler_status.player_port}")
+                self.handler_status.is_connected = 1
+                return
+            except socket.error as msg:
+                print(f"Try: {tries}")
+                print(f"Couldn't connect: {msg}")
+                tries += 1
+                if tries > 5:
+                    self.socket.close()
+                    self.handler_status.is_connected = -1
+                    self.handler_status.to_kill = 1
+                    return
+                time.sleep(1.0)
+
+    def listen(self):
+        self.socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        self.socket.bind((self.handler_status.host_ip, self.handler_status.host_port))
+        self.socket.settimeout(10.0)
+        self.socket.listen(1)
+        print(f"Listening on: {self.handler_status.host_ip}:{self.handler_status.host_port}")
+        try:
+            self.lock.acquire()
+            self.conn, (ip, port) = self.socket.accept()
+            self.socket.settimeout(None)
+            self.lock.release()
+            print(f"Someone connected with me from {ip}:{port}")
+            self.handler_status.is_connected = 1
+        except socket.timeout:
+            self.handler_status.is_connected = -1
+            self.handler_status.to_kill = 1
+
+    def kill(self):
+        if self.handler_status.is_connected == 1:
+            if self.handler_status.is_active_connect == 1:
+                self.socket.shutdown(socket.SHUT_RDWR)
+            else:
+                self.conn.shutdown(socket.SHUT_RDWR)
+                self.conn.close()
+        self.socket.close()
```

Comment:
I'd suggest to change to this. Got to talk about diff between socket and conn.
`
def kill(self):
	try:
		self.socket.shutdown(socket.SHUT_RDWR)
		self.socket.close()
	except SocketError:
		pass
	try:
		self.conn.shutdown(socket.SHUT_RDWR)
		self.conn.close()
	except: #Not sure about expected error prolly SocketError
		pass `

Reply:
Yup, might aswell change that. socket to conn difference described above.